command/whitelist/deployment/params.go

```diff
@@ -0,0 +1,166 @@
+package deployment
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/0xPolygon/polygon-edge/chain"
+	"github.com/0xPolygon/polygon-edge/command"
+	"github.com/0xPolygon/polygon-edge/command/helper"
+	"github.com/0xPolygon/polygon-edge/helper/common"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+const (
+	chainFlag         = "chain"
+	addAddressFlag    = "addAddress"
+	removeAddressFlag = "removeAddress"
+)
+
+var (
+	errInvalidAddressFormat = errors.New("one or more addresses are of invalid format")
+)
+
+var (
+	params = &deploymentParams{}
+)
+
+type deploymentParams struct {
+	// raw addresses, entered by CLI commands
+	addAddressRaw    []string
+	removeAddressRaw []string
+
+	// addresses, converted from raw addresses
+	addAddress    []types.Address
+	removeAddress []types.Address
+
+	// genesis file
+	genesisPath   string
+	genesisConfig *chain.Chain
+
+	// deployment whitelist from genesis configuration
+	whitelist []types.Address
+}
+
+func (p *deploymentParams) initRawParams() error {
+	// convert raw addresses to appropriate format
+	if err := p.initRawAddresses(); err != nil {
+		return err
+	}
+
+	// init genesis configuration
+	if err := p.initChain(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *deploymentParams) initRawAddresses() error {
+	// init slices for addresses to be added and removed
+	p.addAddress = []types.Address{}
+	p.removeAddress = []types.Address{}
```

Comment:
Nitpick here: You know exactly how big these arrays need to be at this point, so you should make them to be that size here

Reply:
Fixed defd863d854e58837f419290e33092e89837936a

Reply:
refactored 4ebfa2bff81d42257da602ea4a026464cbcd646c

Reply:
You can change this new helper to be:
```go
func unmarshallRawAddresses(addresses []string) []types.Address {
	marshalledAddresses := make([]types.Address, len(addresses))

	for indx, address := range addresses {
		marshalledAddresses[indx] = types.StringToAddress(address)
	}

	return marshalledAddresses
}
```

This way you utilize an existing helper for turning a string into an address, and don't have your helper return an error

Reply:
great advice b02f9792d09dbf9dcb4cb343d7283d89f3d62d68

Reply:
Fix lint 336648dbbcca0b525a5f08551e1a97c3c9f26d46